bin/main.rb

```diff
@@ -1,3 +1,78 @@
 #!/usr/bin/env ruby
 
-puts 'Hello World'
+puts '-------------Welcome to Tic Tak Toe-----------------'
+print 'Enter Player 1 name: '
+
+player_one = gets.chomp.capitalize
+
+puts "The Player 1 is #{player_one}"
+
+print 'Enter Player 2 name: '
+
+player_two = gets.chomp.capitalize
+
+puts "The Player 2 is #{player_two}"
+
+puts 'Player 1 Please Select Your Symbol X or O'
+
+player_one_sym = gets.chomp.upcase
+
+while (player_one_sym != 'X') && (player_one_sym != 'O')
+  puts 'Please choose the correct symbol X or O'
+  player_one_sym = gets.chomp.upcase
+end
+
+player_two_sym = if player_one_sym == 'X'
+                   'O'
+                 else
+                   'X'
+                 end
+
+puts "#{player_one} is #{player_one_sym} and #{player_two} is #{player_two_sym}"
+
+current_player = nil
+
+def move_check(move)
+  if (1..9).any?(move)
+    true
+  else
+    false
+  end
+end
+
+def update_board(_move)
+  puts 'Validate move in the Board'
+  puts 'Update Board'
+  puts 'Displays Board'
+end
+
+def check_winner(_player)
+  puts 'check for win condition'
+  puts 'If winner Game Over else Continue Game'
+end
+
+game_on = true
+9.times do |count|
+  game_on = false if count == 8
```

Comment:
While your loop will terminate properly with this setup, a more straight forward approach will be to use the game_on variable to terminate the loop instead of 9.times.
- After each player move, a check for winner or draw is made and if there is a winner or draw, set game_on to false. In that case, your loop should run until game_on is false.